src/rust/ide/span-tree/src/node.rs

```diff
@@ -132,6 +157,18 @@ impl<'a> Ref<'a> {
             None        => Some(self)
         }
     }
+
+    /// Get the node which exaclty matches the given Span:
+    pub fn find_by_span(self, span:&data::text::Span) -> Option<Ref<'a>> {
```

Comment:
We should document behavior when there is more than a single node with the same span.
(or document fact that such situation cannot happen — the question is how section Sides is dealt with)